apps/dashboard/src/components/workflow-editor/steps/in-app/in-app-action.tsx

```diff
@@ -93,15 +94,15 @@ export const InAppAction = () => {
           </DropdownMenuItem>
           <DropdownMenuItem
             onClick={() => {
-              setValue(
-                primaryActionKey,
+              const primaryActionValue = merge(
```

Comment:
when the actions are picked from the dropdown, preserve the previous state, for example: the form is initialized with the `primaryAction` then the user has picked `primaryAction + secondaryAction` we need to preserve the state of the `primaryAction` with what was before

Reply:
there is a video attached that shows this case